activestorage/app/models/active_storage/checksum.rb

```diff
@@ -0,0 +1,128 @@
+# frozen_string_literal: true
+
+class ActiveStorage::Checksum # :nodoc:
+  require "openssl"
+
+  attr_reader :digest, :algorithm
+
+  def initialize(digest, algorithm = nil)
+    @digest, @algorithm = digest, algorithm || :MD5
+  end
+
+  def ==(other)
+    return super unless other.is_a?(ActiveStorage::Checksum)
+    digest == other.digest && algorithm == other.algorithm
+  end
+
+  class << self
+    attr_accessor :crc32_class, :crc32c_class, :crc64_class, :md5_class
```

Comment:
These accessors seem unused?

Reply:
Yes, these can now be removed. They are being using these to memoize the below methods so we dont have to repeatedly make the require call and are not accessed directly.